GH-358: Application State API

Mrnikbobjeff opened the pull request:
### Description of Change ###

Implemented App state as described in Add Application State proposal #358 

### Bugs Fixed ###
None

### API Changes ###
Integration Point: AppInfo:

Added: 
 
```csharp
AppState AppInfo.State { get;  } 
event EventHandler<AppStateChangedEventArgs> AppStateChanged;
```

Changed:
None.

### Behavioral Changes ###
Changed sample project to 7.2 as main project is, to enable readonly struct. If this is not desired it can be removed before merge.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

dend commented:
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Failed to load 1 files, aborting...
 - **[Error]** System.IO.FileNotFoundException: File not found.
File name: &#39;W:\e2uj-log-6780a665\change-list-with-redirection.mapped.tsv&#39;
   at Microsoft.DocAsCode.Build.Engine.Incrementals.ChangeList.Parse(String tsvFile, String baseDir)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()
 - **Line 80**: **[Warning]** Failed to run script W:\e2uj-s\_dependentPackages\ECMA2Yaml\tools\Run.ps1 with exit code -1

<a name=""></a>
### []()
 - **[Error]** Failed to load 1 files, aborting...
 - **[Error]** System.IO.FileNotFoundException: File not found.
File name: &#39;W:\e2uj-log-6780a665\change-list-with-redirection.mapped.tsv&#39;
   at Microsoft.DocAsCode.Build.Engine.Incrementals.ChangeList.Parse(String tsvFile, String baseDir)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()
 - **Line 80**: **[Warning]** Failed to run script W:\e2uj-s\_dependentPackages\ECMA2Yaml\tools\Run.ps1 with exit code -1

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C25%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807252117358752-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=f6JQdx2cCMKrNRcjbjh2QwNq0NlQyd%2BmTn1SBbVRBko%3D&st=2018-07-25T21%3A13%3A12Z&se=2018-08-25T21%3A18%3A12Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

dend commented:
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Failed to load 1 files, aborting...
 - **[Error]** System.IO.FileNotFoundException: File not found.
File name: &#39;W:\hitq-log-a9da8f99\change-list-with-redirection.mapped.tsv&#39;
   at Microsoft.DocAsCode.Build.Engine.Incrementals.ChangeList.Parse(String tsvFile, String baseDir)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()
 - **Line 80**: **[Warning]** Failed to run script W:\hitq-s\_dependentPackages\ECMA2Yaml\tools\Run.ps1 with exit code -1

<a name=""></a>
### []()
 - **[Error]** Failed to load 1 files, aborting...
 - **[Error]** System.IO.FileNotFoundException: File not found.
File name: &#39;W:\hitq-log-a9da8f99\change-list-with-redirection.mapped.tsv&#39;
   at Microsoft.DocAsCode.Build.Engine.Incrementals.ChangeList.Parse(String tsvFile, String baseDir)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()
 - **Line 80**: **[Warning]** Failed to run script W:\hitq-s\_dependentPackages\ECMA2Yaml\tools\Run.ps1 with exit code -1

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C25%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807252118380066-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=sHdRyoxLySTktIRtXEWM1CX0iISRbIsSsmhsYXW3%2BA0%3D&st=2018-07-25T21%3A14%3A17Z&se=2018-08-25T21%3A19%3A17Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

jamesmontemagno commented on Xamarin.Essentials/AppInfo/AppInfo.android.cs:
```diff
@@ -1,11 +1,22 @@
 {
     public static partial class AppInfo
     {
+        static readonly Lazy<AppStateLifecycleListener> appState = new Lazy<AppStateLifecycleListener>(
```
We actually have an AppStateLifecycleListener that is already used in every application. This should be used.

Mrnikbobjeff replied:
Could you perhaps point me there? I have looked around context, and application and can't seem to find it, sorry.

jamesmontemagno commented on Samples/Samples/ViewModel/AppInfoViewModel.cs:
```diff
@@ -15,9 +19,39 @@ public class AppInfoViewModel : BaseViewModel
+
+        public override void OnDisappearing()
+        {
+            Task.Run(async () =>
```
I think that this is maybe better to have at an app wide level perhaps. I think this sample is a bit odd. I don't like random Task.Run, even though I understand why it is there.

Mrnikbobjeff replied:
Okay, would you prefer a static list in the AppInfoViewModel containing all state changes which is subscribed to on application start and never unsubscribed? Otherwise I dont see how we could hook this in the xamarin forms lifecycle, as OnSleep would be called prematurely.

Mrnikbobjeff replied:
@jamesmontemagno I thought more about how app state could be sampled. The current app state while the ui is active is obvious, and on background there is no easy way to display the state. Would my proposal be acceptable? Then I can finish this thing up for good :)

dend commented:
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Failed to load 1 files, aborting...
 - **[Error]** System.IO.FileNotFoundException: File not found.
File name: &#39;W:\jzuo-log-5f4a6806\change-list-with-redirection.mapped.tsv&#39;
   at Microsoft.DocAsCode.Build.Engine.Incrementals.ChangeList.Parse(String tsvFile, String baseDir)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()
 - **Line 80**: **[Warning]** Failed to run script W:\jzuo-s\_dependentPackages\ECMA2Yaml\tools\Run.ps1 with exit code -1

<a name=""></a>
### []()
 - **[Error]** Failed to load 1 files, aborting...
 - **[Error]** System.IO.FileNotFoundException: File not found.
File name: &#39;W:\jzuo-log-5f4a6806\change-list-with-redirection.mapped.tsv&#39;
   at Microsoft.DocAsCode.Build.Engine.Incrementals.ChangeList.Parse(String tsvFile, String baseDir)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument(BuildJsonConfig config, TemplateManager templateManager, String baseDirectory, String outputDirectory, String pluginDirectory, String templateDirectory)
   at Microsoft.DocAsCode.SubCommands.DocumentBuilderWrapper.BuildDocument()
 - **Line 80**: **[Warning]** Failed to run script W:\jzuo-s\_dependentPackages\ECMA2Yaml\tools\Run.ps1 with exit code -1

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C26%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807260217372745-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=awjGvU1nVOxgCZQEcZHs%2F5%2FNvMUWMIYQO4dE0Hu1%2FN8%3D&st=2018-07-26T02%3A13%3A13Z&se=2018-08-26T02%3A18%3A13Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

dend commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AppInfoPage.xaml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/View/AppInfoPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/AppInfoViewModel.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/ViewModel/AppInfoViewModel.cs) | :white_check_mark:Succeeded | |
[Tests/AppInfo_Tests.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Tests/AppInfo_Tests.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.ios.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.shared.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.uwp.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppInfo.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppInfo.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppState.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppState.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C26%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807260313505359-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=Zm0G17FttYo7I1VC%2FhWhrfFob4jfjogqK2p%2F%2B8xjZmg%3D&st=2018-07-26T03%3A09%3A58Z&se=2018-08-26T03%3A14%3A58Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

Redth commented:
build

dend commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AppInfoPage.xaml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/View/AppInfoPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/AppInfoViewModel.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/ViewModel/AppInfoViewModel.cs) | :white_check_mark:Succeeded | |
[Tests/AppInfo_Tests.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Tests/AppInfo_Tests.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.ios.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.shared.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.uwp.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.uwp.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppInfo.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppInfo.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppState.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppState.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C26%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807262257161807-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=lxjTPSGWJZoAbJF%2BK%2F9DaGfRSNVh%2F6cf0t%2F6NoFSt1k%3D&st=2018-07-26T22%3A53%3A57Z&se=2018-08-26T22%3A58%3A57Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

Mrnikbobjeff commented:
Currently this does not forward changes to app background change on android without any subscribers to the event. Should we also hook this in the platform create method and create IComponentCallbacks2 there? This would probably alleviate a lot of confusion!

dend commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AppInfoPage.xaml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/View/AppInfoPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/AppInfoViewModel.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/ViewModel/AppInfoViewModel.cs) | :white_check_mark:Succeeded | |
[Tests/AppInfo_Tests.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Tests/AppInfo_Tests.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.ios.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.shared.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.uwp.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Platform/Platform.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/Platform/Platform.android.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppInfo.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppInfo.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppState.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppState.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C30%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807302001586858-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=iK7Etuf1y6ClVIdTe9%2B1BH0DRltfoAP8fPpXT9l0RIs%3D&st=2018-07-30T19%3A58%3A09Z&se=2018-08-30T20%3A03%3A09Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

dend commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AppInfoPage.xaml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/View/AppInfoPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/AppInfoViewModel.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/ViewModel/AppInfoViewModel.cs) | :white_check_mark:Succeeded | |
[Tests/AppInfo_Tests.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Tests/AppInfo_Tests.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.ios.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.shared.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.uwp.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Platform/Platform.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/Platform/Platform.android.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppInfo.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppInfo.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppState.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppState.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C7%5C30%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201807302048516348-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=3K3ysXGxGRWKwEcfWOpBiLFRKLMDHxQre5yvgIG4HuU%3D&st=2018-07-30T20%3A45%3A10Z&se=2018-08-30T20%3A50%3A10Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

dend commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AppInfoPage.xaml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/View/AppInfoPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/AppInfoViewModel.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/ViewModel/AppInfoViewModel.cs) | :white_check_mark:Succeeded | |
[Tests/AppInfo_Tests.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Tests/AppInfo_Tests.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.ios.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.shared.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.uwp.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Platform/Platform.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/Platform/Platform.android.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppInfo.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppInfo.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppState.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppState.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C1%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808011453037786-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=fQi8QL4zAuyRYx5ylQ9TeZYcMonXPkAPzsqSfVpT5Z4%3D&st=2018-08-01T14%3A49%3A18Z&se=2018-09-01T14%3A54%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

Mrnikbobjeff commented:
@jamesmontemagno I added the componentcallbacks for android to the platform init. This way the app state is always up to date on android. I also sealed the classes in platform. I did not see other instances of sealed classes, I hope that is all right

dend commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AppInfoPage.xaml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/View/AppInfoPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/AppInfoViewModel.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/ViewModel/AppInfoViewModel.cs) | :white_check_mark:Succeeded | |
[Tests/AppInfo_Tests.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Tests/AppInfo_Tests.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.ios.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.shared.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.uwp.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Platform/Platform.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/Platform/Platform.android.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppInfo.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppInfo.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppState.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppState.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C5%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808051427595781-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=z323gbG1GvPNJBnyURG9JWnt1UqewNZaFbwwJYfwNds%3D&st=2018-08-05T14%3A24%3A11Z&se=2018-09-05T14%3A29%3A11Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

dend commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AppInfoPage.xaml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/View/AppInfoPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/AppInfoViewModel.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Samples/Samples/ViewModel/AppInfoViewModel.cs) | :white_check_mark:Succeeded | |
[Tests/AppInfo_Tests.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Tests/AppInfo_Tests.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.ios.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.shared.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/AppInfo/AppInfo.uwp.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/AppInfo/AppInfo.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Platform/Platform.android.cs](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/Xamarin.Essentials/Platform/Platform.android.cs) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-android.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-android.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-ios.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-ios.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials-uwp.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials-uwp.xml) | :white_check_mark:Succeeded | |
[docs/en/FrameworksIndex/xamarin-essentials.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/FrameworksIndex/xamarin-essentials.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppInfo.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppInfo.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppState.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppState.xml) | :white_check_mark:Succeeded | |
[docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/Xamarin.Essentials/AppStateChangedEventArgs.xml) | :white_check_mark:Succeeded | |
[docs/en/index.xml](https://github.com/Mrnikbobjeff/Essentials/blob/feature/FinalAppState/docs/en/index.xml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C10%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808101514059748-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=UiJ9m4XZpDojWXG0nXGXh8JOmVESUPLOp8o%2FYAc5J8c%3D&st=2018-08-10T15%3A10%3A19Z&se=2018-09-10T15%3A15%3A19Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

mattleibow commented:
How does this work with the UWP multi-window and mini-window/content-in-content (ie: Netflix/video player)

https://docs.microsoft.com/en-us/windows/uwp/design/layout/show-multiple-views

jamesmontemagno commented:
I am not super comfortable with this feature and think we should put it on the back burner for now.

Mrnikbobjeff commented:
@mattleibow Do you mean when the App itself is a mini Window or when it is "covered" by a mini-wi die. I'll do some investigating,  more updates tomorrow.

Mrnikbobjeff commented:
I promised updates today, sadly the newest VS 15.8 build shot all my emulators and my connection to the mac build host. Until I fixed VS all investigation from my side is on hold. I will call back once I fixed this mess

Mrnikbobjeff commented:
@mattleibow You are correct, there were indeed hidden problems with multi window/in window:
UWP: No problems with window in window, multi window fired the event again even if the current state did not change. Fixed by caching the last value and only raising event if the state changed.
Android: No problem with multi window, in window suffered from the same problem as described above. The fix is in shared code, so this worked there as well. I don't have a mac/iPhone with me, as soon as I am back at home I will check iOS.

dend commented:
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-)

<a name=""></a>
### []()
 - **[Error]** Cannot sync git repo to specified state. Please make sure the pull request is mergeable. Details: Git merge results in conflict, repository remote url: https://github.com/xamarin/Essentials, working directory: W:\lfya-s, target branch: master, target branch commit: 459798ff19a59f8fe883a57e6e4fbdb70f5b228c, source branch: 7cc8a3093c463ded66e89819d67a6c08/feature/FinalAppState, source branch commit: .

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C17%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201808171844500379-412%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=QMH4EQzi8C7ajdUAgyKFpR%2BhWzkqeWNglyJXQHAqa%2Bs%3D&st=2018-08-17T18%3A39%3A59Z&se=2018-09-17T18%3A44%3A59Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

Mrnikbobjeff commented:
I had no problems on iOS with the aforementioned scenario. What I still have to test is when the app is the mini window. I am not sure on how we would want this to be handled, probably as foreground? I have never implemented this mini window feature, so it will take some time to create repros for all 3 OS.

jamesmontemagno commented:
After examining this API we have come to the conclusion that it is not a good fit for Essentials as it can not be properly abstracted across each platform. I want to thank you for your effort in this PR and comments on the issue and talking through all of the concerns.
